src/core/query/json_woql.pl

```diff
@@ -759,16 +760,15 @@
     ).
 json_type_to_woql_ast('GroupBy',JSON,WOQL,Path) :-
     _{group_by : Spec,
-      template : Obj,
+      template : Template,
       query : Query,
       grouped : Collector
      } :< JSON,
     variable_list(Spec,WSpec),
-    variable_list(Obj,Template),
-    % Add only one variable if only one in template
-    (   Template = [V]
-    ->  WObj = V
-    ;   WObj = Template),
+    % Backwards compat with "variable list"
+    (   variable_list(Template,WObj)
+    ->  true
+    ;   json_to_woql_ast(Template,WObj,[template|Path])),
```

Comment:
This is too abstract - since we can't actually deal with an arbitrary query here. Instead we need a datatype to ast conversion predicate, and the cases of list, dictionary, variable, node, etc. need to be handled there.